libs/queue/SPECIFICATION.md

```diff
@@ -0,0 +1,35 @@
+# Overview
+
+This document provides an overview of the Queue library.
+
+It outlines the use cases, i.e. specification, and describes how to implement the library.
+
+## Use Cases
+
+The Queue library can be used as a traditional queue or first in first out data structure.
```

Comment:
I don't know what this sentence is trying to tell me.

Should I assume that there are non-traditional queues, which is the traditional one? The end of the sentence mentions the FIFO context which is fine but this is the "use case" section and this does not describe a use case - it describes how the implementation functions.

I think the confusion is on the terminology of "use case".
A use case for a business could be to retrieve the first data packet that it receives i.e. FIFO (UDP in networking for example). The queue then meets that _requirement_ because it provides the functionality and thus the overall "use case" for the business is satisfied.

Reply:
Got it, I will commit a rewritten version of this.